### PR TITLE
feat: add button to easily clear filter search bars (#3612)

### PR DESCRIPTION
**Pull request by frano-m**
- #3612 

### Reviewers
**Functional:** 
@tihuan 

---

## Changes
- Added clear button to filter search as per [mock](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?node-id=4261%3A37785&t=BHyMIN9m4P1kvKj6-4) and implemented as per [slack discussion](https://clevercanary.slack.com/archives/C02CF7MQPGV/p1669055439987369?thread_ts=1668452415.971099&cid=C02CF7MQPGV).

### CODE REVIEW

File: frontend/src/components/common/Filter/components/FilterSearch/index.tsx [hunk 1]
```diff
@@ -13,13 +15,38 @@ export default function FilterSearch({
...
+      inputRef={inputRef}
       leftIcon={IconNames.SEARCH}
       onChange={(changeEvent) => setSearchValue(changeEvent.target.value)}
       placeholder="Search"
+      rightElement={
+        searchValue ? (
+          <ClearIconButton
+            onClick={onClearSearch}
+            {...{
+              // TODO(cc) move this back to explicit prop={value} after upgrading SDS to enable type checking again
```
Comment (frano-m) [inline frontend/src/components/common/Filter/components/FilterSearch/index.tsx:41]: Newer versions require SDS prop `sdsSize` for IconButton.

Comment (tihuan) [reply frontend/src/components/common/Filter/components/FilterSearch/index.tsx:41]: Thanks for the comment! I'm working on upgrading to the latest SDS now. Hopefully can get out a PR next week 🚀 !

Comment (tihuan) [reply frontend/src/components/common/Filter/components/FilterSearch/index.tsx:41]: I'll make sure to update the code here in my SDS upgrade PR too 👍

**Review by tihuan — APPROVED** [review]
LGTM! Thanks so much for the nice touch of refocusing the input field, @frano-m 👍 ✨

File: frontend/src/components/common/Filter/components/FilterSearch/index.tsx (hunk 1 above)
Comment (frano-m) [reply frontend/src/components/common/Filter/components/FilterSearch/index.tsx:41]: Thanks so much @tihuan! So pumped for the latest SDS - it looks 🤩 !

**Pull request merged**